creusot/tests/should_succeed/iterators/06_map_precond.coma

```diff
@@ -40,9 +37,7 @@ module T_alloc__alloc__Global
   
   function any_l (_ : 'b) : 'a
   
-  let rec t_Global (input:t_Global) (ret  )= any
-    [ good -> {C_Global  = input} (! ret) | bad -> {C_Global  <> input} {false} any ]
-    
+  let rec t_Global (input:t_Global) (ret  )= any [ good -> {C_Global  = input} (! ret) ] 
```

Comment:
@paulpatault additionally, if you can confirm its safe to remove the `bad` branch for product typesl ike this it would be useful.

Reply:
I don't see where is the product type, there is only one constructor in `type t_Global  = | C_Global` or I miss something ?